### PR TITLE
Feature/install events

### PR DESCRIPTION
**Pull request by Goopil**
implements pre & post install events has discussed in #268 

## usage
```php
// App/Providers/AppServiceProvider.php

use Nwidart\Modules\Events\ModuleIsInstalling;
use Nwidart\Modules\Events\ModuleWasInstalled;
...


    public function boot()
    {
        \Event::listen(ModuleIsInstalling::class, function (ModuleIsInstalling $event) {
            $module = $event->getModule();
            $console = $event->getConsole();
        });

        \Event::listen(ModuleWasInstalled::class, function (ModuleWasInstalled $event) {
            $module = $event->getModule();
            $console = $event->getConsole();
        });
    }
```

### CODE REVIEW

**Comment by nWidart** [comment]
These events don't need the console, repository, module. Just the module name is enough.

File: src/Process/Installer.php [hunk 1]
```diff
@@ -57,14 +59,14 @@ class Installer
...
      */
     public function __construct($name, $version = null, $type = null, $tree = false)
     {
-        $this->name = $name;
+        $this->name    = $name;
         $this->version = $version;
-        $this->type = $type;
-        $this->tree = $tree;
+        $this->type    = $type;
+        $this->tree    = $tree;
```
Comment (nWidart) [inline src/Process/Installer.php]: Code style issue here.

**Comment by nWidart** [comment]
You're also missing blank lines at end of files. https://insight.sensiolabs.com/projects/25320a08-8af4-475e-a23e-3321f55bf8d2/analyses/187

File: src/Process/Installer.php (hunk 1 above)
Comment (Goopil) [inline src/Process/Installer.php]: You are right this is php storm doing it one thing here.

**Comment by Goopil** [comment]
The console should be availlable if you want to prompt a choice. Don't you think ? Ok for the code Style.

File: src/Events/ModuleBaseEvent.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    /**
+     * The console command instance.
+     *
+     * @var Command
+     */
+    protected $console;
+
+    public function __construct($name, Repository $repository, Command $console)
+    {
+        $this->module = $repository->get($name);
```
Comment (nWidart) [inline src/Events/ModuleBaseEvent.php]: Please don't use the repository to get the name, when you have the name right here. This is for performance reasons. The repository isn't needed in the event.

Comment (Goopil) [inline src/Events/ModuleBaseEvent.php]: This is not to have the name but the module itself. Should i do it in the installer instead ?

Comment (nWidart) [inline src/Events/ModuleBaseEvent.php]: I don't think the event needs the module instance. As I mentioned in the issue, the name is enough.

File: src/Process/Installer.php [hunk 3]
```diff
@@ -134,10 +134,10 @@ public function run()
     {
         $process = $this->getProcess();
         $app = $this->console->getLaravel();
+        $module = $this->repository->get($this->getModuleName());
```
Comment (nWidart) [inline src/Process/Installer.php]: Why? The **name** is enough.

Comment (Goopil) [inline src/Process/Installer.php]: updated if you still want it

**Pull request closed without merging**